lib/linter.js

```diff
@@ -50,6 +49,40 @@ const MAX_AUTOFIX_PASSES = 10;
 //------------------------------------------------------------------------------
 
 /**
+ * Creates an object which can listen for and emit events.
+ * This is similar to the EventEmitter API in Node's standard library, but it has a few differences.
+ * The goal is to allow multiple modules to attach arbitrary listeners to the same emitter, without
+ * letting the modules know about each other at all.
+ * 1. It has no special keys like `error` and `newListener`, which would allow modules to detect when
+ * another module throws an error or registers a listener.
+ * 2. It calls listener functions without any `this` value. (`EventEmitter` calls listeners with a
+ * `this` value of the emitter instance, which would give listeners access to other listeners.)
+ * 3. Events can be emitted with at most 3 arguments.
+ * @returns {{on: Function, emit: Function, eventNames: Function}} An emitter
```

Comment:
I'd love to see this document in a JSDoc typedef, but this is nowhere near a blocker.